### ISSUE TITLE
Automatically define activities based on MassTransit Messages

### ISSUE THREAD

**Issue by sfmskywalker** [elsa-workflows/elsa-core#112]
Currently, activity definitions are harvested by scanning the assemblies for types implementing IActivity. 

If we abstract the activity discovery process, we can provide other sources of activity definitions. For example, in a MassTransit powered application, activities could by yielded based on message classes, automatically making these messages available as activities.

Similarly, one could provide an Open API specification (Swagger), and a host of activities will be available to the user to interact with the API in question.

**Comment by Andale-online** [comment]
> Currently, activity definitions are harvested by scanning the assemblies for types implementing IActivity.
Is it so, sir?
> 
> If we abstract the activity discovery process, we can provide other sources of activity definitions. 
My talking - we need more code / assembly flexibility :-P

**Comment by sfmskywalker** [comment]
It is so - but not in the same way that I think you are suggesting we do by loading assemblies - that "plugin" loading ability is to allow users to drop in assemblies that can be picked up & loaded into the application dynamically at runtime.

I'm merely speaking of resolving an `IEnumerable<IActivity>` - which would be done way after any assemblies have been loaded (including the dynamically loaded ones).

In other words, there won't be any activity harvesters that use dynamic assembly loading. The way I see it is that the plugin infrastructure has nothing to do with Elsa - it's just a feature you implement in your application using whichever method you prefer. Elsa, and other libraries, sit on top of that, being unaware of any types loaded dynamically.

**Comment by Andale-online** [comment]
No, sorry, didn't want to take over this issue. Just wanted to clarify something for myself. I'll put up a new enhancement issue to describe what I was after.
Greetings
